### PR TITLE
[BE] [EZ] Allow linux-build workflows to run on the default runner type

### PR DESCRIPTION
**Pull request by ZainRizvi**
Replace usage of `runner` with the new `runner_prefix` input, which allows the workflows to use the default runner type (linux.2xlarge) specified by the reusable workflow.

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]

**Review by clee2000 — COMMENTED** [review]
q: how did this work before?  It seems like the runner type would have included linux.2xlarge twice?

**Comment by jeanschmidt** [comment]
aren't we shooting ourselves in the foot by making those super specific changes that we'll need to repeat them every time we want to make a migration?

I believed that we were in a situation where we could have some flexibility already and could avoid those changes.

**Review by jeanschmidt — APPROVED** [review]

File: .github/workflows/periodic.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
     uses: ./.github/workflows/_linux-build.yml
     needs: get-label-type
     with:
-      runner: "${{ needs.get-label-type.outputs.label-type }}amz2023.linux.2xlarge"
+      runner_prefix: "${{ needs.get-label-type.outputs.label-type }}amz2023."
```
Comment (malfet) [inline .github/workflows/periodic.yml:80]: Why repeated `needs.get-label-type.outputs.label-type` can not be refactored away?

**Review by malfet — APPROVED** [review]

File: .github/workflows/periodic.yml (hunk 1 above)
Comment (ZainRizvi) [reply .github/workflows/periodic.yml:80]: how would one do that?  That prefix today contains either the "lf." prefix or an empty string and is set by the get-label-type job to figure out if the jobs should run in the LF fleet or Meta fleet

**Comment by ZainRizvi** [comment]
> q: how did this work before? It seems like the runner type would have included linux.2xlarge twice?

No, before we set the variable "runner" (which defaults to "linux.2xlarge"). Now we set a new variable "runner_prefix"

**Comment by ZainRizvi** [comment]
> aren't we shooting ourselves in the foot by making those super specific changes that we'll need to repeat them every time we want to make a migration?
> 
> I believed that we were in a situation where we could have some flexibility already and could avoid those changes.

The idea is that eventually we'll only need to update the runner determinator (get-label-type) to pull in the fleet level changes.  However, to change the base runner type (e.g. linux.2xlarge) we would only need to make the change in one place (_linux-build.yml), like how it used to be.

I basically see this PR as paying off tech debt from our migrations. 

Though we will need one more round of changes for the amz2023 cleanup this week.

**Pull request closed without merging**